tests/run.sh

```diff
@@ -1,3 +1,46 @@
 #!/bin/sh
 
-nvim --headless -u tests/init.lua -c "PlenaryBustedDirectory tests/ {minimal_init = 'tests/init.lua', sequential = true}"
+GITHUB="https://github.com"
+GITHUB_PLENARY="$GITHUB/nvim-lua/plenary.nvim"
+GITHUB_TELESCOPE="$GITHUB/nvim-telescope/telescope.nvim"
+GITHUB_WHICHKEY="$GITHUB/nvim-telescope/folke/which-key.nvim"
+
+SCRIPT_FILE=$(readlink -f "${BASH_SOURCE[0]}")
+REPO_DIR=$(dirname "$(dirname "$script_file")")
+
+TEST_MINIMAL_DIR="$REPO_DIR/.tests/minimal/site/pack/deps/start"
+TEST_TELESCOPE_DIR="$REPO_DIR/.tests/telescope/site/pack/deps/start"
+TEST_LSP_DIR="$REPO_DIR/.tests/lsp/site/pack/deps/start"
+TEST_WHICHKEY_DIR="$REPO_DIR/.tests/whichkey/site/pack/deps/start"
+
+# The Tests Runner runs tests for all the environments
+# TODO: Support running subset of tests
+
+clone() {
+  repo=$1
+  dest=$2
+  if [ ! -d "$dest" ] ; then
+    git clone --depth 1 "$repo" "$dest"
+  else
+    echo "$1 already cloned"
+  fi
+}
+
+mkdir -p $TEST_MINIMAL_DIR
+clone $GITHUB_PLENARY "$TEST_MINIMAL_DIR/plenary.nvim"
+nvim --headless -u tests/environments/minimal.lua -c "PlenaryBustedDirectory tests/ {minimal_init = 'tests/environments/minimal.lua', sequential = true}"
+
+# mkdir -p $TEST_LSP_DIR
+# git clone --depth 1 $GITHUB_PLENARY "$TEST_LSP_DIR/plenary.nvim"
+# nvim --headless -u tests/environments/lsp.lua -c "PlenaryBustedDirectory tests/ {minimal_init = 'tests/environments/lsp.lua', sequential = true}"
```

Comment:
Is the idea then to split by directory?

Reply:
The environments are split into directories, I'm considering 2 approaches for splitting the tests:

* Run only the tests for the plugin environment. ie: 
```
nvim -u tests/environments/lsp.lua -c "PlenaryBustedDirectory tests/lsp/
nvim -u tests/environments/telescope.lua -c "PlenaryBustedDirectory tests/telescope/
```

* Run all the tests for all the plugin environments
The tests would contain guard statements to avoid including tests according to the feature availability